cmd/rookflex/cmd/mount.go

```diff
@@ -174,3 +181,51 @@ func mount(client *rpc.Client, mounter *k8smount.SafeFormatAndMount, globalVolum
 	}
 	return err
 }
+
+func mountCephFS(client *rpc.Client, opts *flexvolume.AttachOptions) error {
+
+	log(client, fmt.Sprintf("mounting ceph filesystem %s on %s", opts.VolumeName, opts.MountDir), false)
+
+	if opts.ClusterName == "" {
+		return fmt.Errorf("Rook: Attach filesystem %s failed: cluster is not provided", opts.VolumeName)
+	}
+
+	// Get client access info
+	var clientAccessInfo model.ClientAccessInfo
+	err := client.Call("FlexvolumeController.GetClientAccessInfo", opts.ClusterName, &clientAccessInfo)
+	if err != nil {
+		log(client, fmt.Sprintf("Attach filesystem %s on cluster %s failed: %v", opts.VolumeName, opts.ClusterName, err), true)
+		return fmt.Errorf("Rook: Attach filesystem %s on cluster %s failed: %v", opts.VolumeName, opts.ClusterName, err)
+	}
+	devicePath := strings.Join(clientAccessInfo.MonAddresses, ",") + ":/"
+	options := []string{fmt.Sprintf("name=%s", clientAccessInfo.UserName), fmt.Sprintf("secret=%s", clientAccessInfo.SecretKey), fmt.Sprintf("mds_namespace=%s", opts.VolumeName)}
+
+	mounter := getMounter()
+	// Perform a bind mount to the full path to allow duplicate mounts of the same volume. This is only supported for RO attachments.
+	err = redirectStdout(
+		client,
+		func() error {
+
+			notMnt, err := mounter.Interface.IsLikelyNotMountPoint(opts.MountDir)
+			if err != nil && !os.IsNotExist(err) {
+				return err
+			}
+			if !notMnt {
+				return nil
+			}
+			os.MkdirAll(opts.MountDir, 0750)
+
+			err = mounter.Interface.Mount(devicePath, opts.MountDir, "ceph", options)
+			if err != nil {
+				// cleanup upon failure
+				util.UnmountPath(opts.MountDir, mounter.Interface)
+				return fmt.Errorf("failed to mount filesystem %s to %s with monitor %s and options %v: %+v", opts.VolumeName, opts.MountDir, opts.MountDir, options, err)
```

Comment:
I think the third var should be the list of monitors and not the `opts.MountDir` again, right?

Reply:
Nice catch. Will change it.